editor/plugins/configuration_info_editor_plugin.cpp

```diff
@@ -0,0 +1,187 @@
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "configuration_info_editor_plugin.h"
+
+#include "editor/editor_configuration_info.h"
+#include "editor/editor_string_names.h"
+#include "scene/gui/grid_container.h"
+#include "scene/gui/rich_text_label.h"
+
+// Inspector controls.
+
+void ConfigurationInfoList::_update_content() {
+	if (!object) {
+		hide();
+		return;
+	}
+
+	const LocalVector<ConfigurationInfo> config_infos = EditorConfigurationInfo::get_configuration_info(object);
+	if (config_infos.is_empty()) {
+		hide();
+		return;
+	}
+
+	title_label->set_text(vformat(TTRN("%d Configuration Info", "%d Configuration Infos", config_infos.size()), config_infos.size()));
+
+	const Color &warning_color = get_theme_color(StringName("warning_color"), EditorStringName(Editor));
+	const Color &error_color = get_theme_color(StringName("error_color"), EditorStringName(Editor));
```

Comment:
SNAME descrpition says it should only be used where performance is needed, and not to use when in doubt. So since it updates rarely, I didn't use it here.

But I can switch to using strings and having it convert implicitly, if that is preferred style.